### PR TITLE
#10001 edit gulpfile for travis build

### PR DESCRIPTION
**Pull request by johnkingzy**
#### What does this PR do?
Edit Gulpfile for a successful travis build
#### Description of Task to be completed?
N/A
#### How should this be manually tested?
run gulp test
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: src/js/app.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        priority: 'danger',
+        title: '',
+        message: msg
+      });
+    };
+
+    $scope.createIndex = (selectedFile) => {
+      if (selectedFile === null || typeof selectedFile === 'undefined') {
+        $scope.msgDisplay('danger', 'Select a File to Create Index');
+        // $("#myModal").modal('show');
```
Comment (oyinda-subair) [inline src/js/app.js]: Remove commented code(Dead code)

File: src/js/app.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  /** @function init
+   * @return {null} null
+  */
+  const table = () => (input, arr) => {
+    if (arr.includes(input)) {
+      return 'check';
+    }
+    return 'cross';
+  };
+  // };
```
Comment (oyinda-subair) [inline src/js/app.js]: Same here (Dead Code), Also your JSDoc return statement is null, but you are returning string in your function block

File: .coveralls.yml [hunk 3]
```diff
@@ -1 +1 @@
-repo_token: cXzFmBBM5HRvU7Rz7lxNz1KqWSBH0UtN6
\ No newline at end of file
+repo_token: gXgeoORGKcIyxW5hqLaD17QIkjTdjozWH
```
Comment (oagboola) [inline .coveralls.yml]: you shouldn't put up credentials publicly on gh, this should be in .env file

File: src/js/app.js (hunk 1 above)
Comment (johnkingzy) [reply src/js/app.js]: Thanks this has been updated

File: .coveralls.yml (hunk 3 above)
Comment (johnkingzy) [reply .coveralls.yml]: Thanks this has been updated

File: src/js/app.js (hunk 2 above)
Comment (johnkingzy) [reply src/js/app.js]: Thanks this has been updated

File: src/js/app.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    $scope.home = () => {
+      $scope.showSearch = false;
+      $scope.showIndex = false;
+      $scope.welcome = true;
+    };
+
+    /** @function init
+     * @param {uploadedFile} uploadedFile
+   * @return {null} null
+  */
```
Comment (iAmao) [inline src/js/app.js:108]: Indent your jsdocs properly

File: src/js/app.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    function uploadFile(uploadedFile) {
+      const files = uploadedFile.target.files;
+      $scope.replaceFiles = [];
+      const keys = Object.keys(files);
```
Comment (iAmao) [inline src/js/app.js:112]: I don't really see the point of putting  Object.keys(files) in another variable

File: src/js/app.js [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+          .catch((error) => {
+            $scope.msgDisplay('danger', error);
+            $scope.$apply();
+          });
+      });
+    }
+    // We can attach the `fileselect` event to all file inputs on the page
+    /** @function init
+   * @return {null} null
+  */
```
Comment (iAmao) [inline src/js/app.js:137]: Properly indent your jsdocs

File: src/js/app.js [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+            });
+          })
+          .catch((error) => {
+            $scope.msgDisplay('danger', error);
+            $scope.$apply();
+          });
+      });
+    }
+    // We can attach the `fileselect` event to all file inputs on the page
+    /** @function init
```
Comment (iAmao) [inline src/js/app.js:135]: init is not a function

File: src/js/app.js [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+  */
+    function Init() {
+      const fileselect = document.getElementById('fileselect');
+      fileselect.addEventListener('change', uploadFile, false);
+    }
+    if (window.File && window.FileList && window.FileReader) {
+      Init();
+    }
+  }
+  /** @function table
```
Comment (iAmao) [inline src/js/app.js:146]: what does table do?

File: src/js/app.js [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+   * @return {String} string
+  */
```
Comment (iAmao) [inline src/js/app.js:148]: Where are your params?

File: src/js/invertedIndex.js [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+   * @createIndex method
+   * @param {string} fileName
+   * @param {Object} fileContent
+   * @returns {boolean} true - index created successfully
+   * create index of the fileName
+   */
+  createIndex(fileName, fileContent) {
+    this.indices[fileName] = this.indices[fileName] || {};
+    const totalBooks = fileContent.length;
+    for (let bookIndex = 0; bookIndex <
```
Comment (iAmao) [inline src/js/invertedIndex.js:26]: You can use a forEach here instead of a for loop

File: src/js/invertedIndex.js [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+      .replace(/[[|\]]/g, '')
+      .toLowerCase()
+      .trim()
+      .split(/\b\s+(?!$)/);
+    value = value.filter(elem => elem !== '');
+    return value;
+  }
+}
+/* eslint-disable no-unused-vars */
+const invertedIndex = new InvertedIndex();
```
Comment (andela-iamao) [inline src/js/invertedIndex.js]: Why?

File: src/js/invertedIndex.js (hunk 10 above)
Comment (johnkingzy) [reply src/js/invertedIndex.js:26]: Thanks this has been updated

File: src/js/app.js (hunk 9 above)
Comment (johnkingzy) [reply src/js/app.js:148]: thanks this has been updated

File: src/js/app.js (hunk 8 above)
Comment (johnkingzy) [reply src/js/app.js:146]: thanks this has been updated

File: src/js/app.js (hunk 6 above)
Comment (johnkingzy) [reply src/js/app.js:137]: thanks this has been updated

File: src/js/app.js (hunk 7 above)
Comment (johnkingzy) [reply src/js/app.js:135]: thanks this has been updated

File: src/js/app.js (hunk 5 above)
Comment (johnkingzy) [reply src/js/app.js:112]: thanks this has been updated

File: src/js/app.js (hunk 4 above)
Comment (johnkingzy) [reply src/js/app.js:108]: thanks this has been updated

**Pull request merged**